be/src/exec/pipeline/exchange/exchange_parallel_merge_source_operator.h

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "exec/pipeline/source_operator.h"
+#include "exec/sorting/merge_path.h"
+
+namespace starrocks {
+class SortExecExprs;
+namespace pipeline {
+
+// The number of sender may not equal to the degree of parallelism at the receiver side.
+//
+//        Sender                                                 Receiver
+//
+// +----------------------+       Network        +-------------------------------------+
+// | ExchangeSinkOperator | -------------------> | ExchangeParallelMergeSourceOperator | ------------> output streams
+// +----------------------+                      +-------------------------------------+
+// +----------------------+       Network        +-------------------------------------+
+// | ExchangeSinkOperator | -------------------> | ExchangeParallelMergeSourceOperator | no output stream
+// +----------------------+                      +-------------------------------------+
+// +----------------------+       Network        +-------------------------------------+
+// | ExchangeSinkOperator | -------------------> | ExchangeParallelMergeSourceOperator | no output stream
+// +----------------------+                ----->+-------------------------------------+
+// +----------------------+      ---------/
+// | ExchangeSinkOperator |-----/
+// +----------------------+
+//
+// There will be as many ExchangeParallelMergeSourceOperator as degree of parallelism, and
+// all the ExchangeParallelMergeSourceOperators will be involved in the parallel merge processing,
+// and only one ExchangeParallelMergeSourceOperator, the operator with driver_sequence = 0 for simplicity,
+// will output data.
```

Comment:
Can I ask where controls this `the operator with driver_sequence = 0 for simplicity, will output data`? 
I don't see it in the `has_output` method.

Reply:
ExchangeParallelMergeSourceOperator::has_output is to trigger the pull_chunk, it has nothing todo with the finally output data.

You can check this logic at `MergePathCascadeMerger::_fetch_chunk`

```cpp
if (parallel_idx != 0) {
    finished = true;
    return;
}
```